### ISSUE TITLE
proc_func does not proceed

### ISSUE THREAD

**Issue by KobaCemal** [MICA-MNI/micapipe#116]
Dear developers,

I am trying to run micapipe on my datasets. I manage to get through -proc_structural, -proc_surf, -post_structural, and -GD. However, I face an error for both of my datasets at -proc_func step. The error is below:
![mpipe_error](https://github.com/MICA-MNI/micapipe/assets/45714553/bd0c02b2-8a16-4773-9cc0-ddbdcd61c4ae)

My code is here: 
```
for sub in 001  
do
sudo docker run -ti --rm \
    -v ${bids}:/bids \
    -v ${out}:/out \
    -v ${fs_lic}:/opt/licence.txt \
    micalab/micapipe:v0.2.3 \
        -bids /bids -out /out -fs_licence /opt/licence.txt -threads 10 \
        -sub $sub \
         -proc_func -mainScanStr rest_bold
done

```

The datasets are in BIDS format. At least they pass the validator. And they both have the RepetitionTime info in their json files. I think the structural part would not work if there was a problem with the format. 

My guess is that the problem comes from the line 85 of the proc_func (can be seen at the top of the screenshot). I would work on it but I cannot figure out how to edit a file in a docker image.

This is the folder structure of one if the datasets: 
├── sub-001
│   ├── anat
│   │   ├── sub-001_T1w.json
│   │   └── sub-001_T1w.nii.gz
│   └── func
│       ├── sub-001_task-rest_bold.json
│       └── sub-001_task-rest_bold.nii.gz
├── sub-002
│   ├── anat
│   │   ├── sub-002_T1w.json
│   │   └── sub-002_T1w.nii.gz
│   └── func
│       ├── sub-002_task-rest_bold.json
│       └── sub-002_task-rest_bold.nii.gz
├── sub-003
│   ├── anat
│   │   ├── sub-003_T1w.json
│   │   └── sub-003_T1w.nii.gz
│   └── func
│       ├── sub-003_task-rest_bold.json
│       └── sub-003_task-rest_bold.nii.gz

Since it maybe related, I will also add this recent unusual error when I try to run post_structural as well:

-------------------------------------------------------------

[ ERROR ]..... Subject sccb01 doesn't have a mri/ribbon.mgz: re-run -proc_surf
 /out/fastsurfer/sub-sccb01/mri/ribbon.mgz

-------------------------------------------------------------

Basically, it cannot generate the ribbon for some subjects and I need to rerun the proc_structural. Speaking of which, if you can tell how to run micapipe_cleanup on docker, that would be great because I could not follow it on the documentation. 

Thank you for your help. 

Best,
Cemal

**Comment by KobaCemal** [comment]
Hello,

I solved the problem. The issue was wrong specification of -mainScanStr string. I fixed it and now it works. 
Thanks!

**Comment by yanfeifei2008** [comment]
> Hello,
> 
> I solved the problem. The issue was wrong specification of -mainScanStr string. I fixed it and now it works. Thanks!

Hi Cemal,
I'm writing because I've encountered the same problem you described above. Could you please share some detailed advice on how you managed to solve it?
Any guidance you could provide would be greatly appreciated.
Thanks, 
Best regards, 
Yan

**Comment by KobaCemal** [comment]
> > Hello,
> > I solved the problem. The issue was wrong specification of -mainScanStr string. I fixed it and now it works. Thanks!
> 
> Hi Cemal, I'm writing because I've encountered the same problem you described above. Could you please share some detailed advice on how you managed to solve it? Any guidance you could provide would be greatly appreciated. Thanks, Best regards, Yan

Hello Yan, 

The code that worked for me is here: 
```

for scan in "sub-016"
do 
sub=$(echo $scan | cut -d "_" -f 1 | cut -d "-" -f2)
ses=$(echo $scan | cut -d "_" -f 2 | cut -d "-" -f2)
 
singularity run -B $SCRATCH  $SCRATCH/micalab_micapipe_v0.2.3-2024-01-18-4ee63843ce22.simg \
    -bids $SCRATCH/datasets/baltimore \
    -out $SCRATCH/datasets/baltimore/derivatives \
    -tmpDir $SCRATCH/tmp \
    -ses $ses \
    -threads 16\
    -fs_licence $SCRATCH/license.txt \
    -sub $sub  -proc_structural -proc_surf -post_structural -atlas glasser-360 -GD -proc_func -mainScanStr task-rest_bold -noFIX -NSR -QC_subj
done
 
```

I was using the task/rest string wrong (rest_bold). Turns out you need to give the prefix as well: task-rest_bold. I hope this is similar to what you have. 

See how I removed the "sub" prefix from participant IDs. Micapipe doesnt like that either.
